src/main/java/ru/skypro/homework/controller/AdController.java

```diff
@@ -0,0 +1,87 @@
+package ru.skypro.homework.controller;
+
+import jakarta.validation.Valid;
+import lombok.AllArgsConstructor;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import ru.skypro.homework.dto.*;
+import ru.skypro.homework.service.AdService;
+import ru.skypro.homework.service.CommentService;
+
+@Validated
+@RestController
+@RequestMapping(path = "/ads")
+@AllArgsConstructor
+public class AdController {
+
+    private final AdService adService;
+    private final CommentService commentService;
+
+    @GetMapping
+    public ResponseEntity<Ads> getAds() {
+        return ResponseEntity.ok(adService.getAllAds());
+    }
+
+    @PostMapping(consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+    public ResponseEntity<Ad> addAd(@RequestBody @Valid CreateOrUpdateAd properties, @RequestBody MultipartFile image) {
```

Comment:
```suggestion
    public ResponseEntity<Ad> addAd(@RequestPart @Valid CreateOrUpdateAd properties, @RequestPart MultipartFile image) {
```
RequestBody не может 2 применяться, т.к. тело запроса одно. В данном случае тип контента тела запроса multipart/form-data, поэтому принимать надо части запроса